cmd/oras/internal/option/remote.go

```diff
@@ -273,6 +283,16 @@ func (opts *Remote) NewRepository(reference string, common Common) (repo *remote
 	if repo.Client, err = opts.authClient(hostname, common.Debug); err != nil {
 		return nil, err
 	}
+
+	if opts.distributionSpec.referrersAPI == nil || *opts.distributionSpec.referrersAPI {
+		once := sync.Once{}
+		repo.HandleWarning = func(warning remote.Warning) {
+			once.Do(func() {
+				warn(warning.Text)
+			})
+		}
```

Comment:
I am confused with the `once`. Is that for deduplication? If so, it is probably wrong.

Consider that the server may send out:
```
Warning: foo
Warning: bar
Warning: foo
```
The client should print
```
Warning: foo
Warning: bar
```
The last warning of `foo` is deduplicated.

Reply:
Thanks for reminding, I am not aware of that. Is the warning deduplicated based on the warning full message?

Reply:
discussed offline with @Wwwsylvia and resolved this comment